src/rgw/rgw_rest.cc

```diff
@@ -655,7 +655,9 @@ void end_header(struct req_state* s, RGWOp* op, const char *content_type,
       s->formatter->close_section();
     }
     s->formatter->output_footer();
-    dump_content_length(s, s->formatter->get_len());
+    if (s->obj_size) {
```

Comment:
you have here two content lengths headers,
it is because you add the content length even in case of an error see my comment on rgw_rest_swift.cc